app/scripts/controllers/swaps.js

```diff
@@ -136,13 +134,6 @@ export default class SwapsController {
     this.indexOfNewestCallInFlight = 0;
 
     this.ethersProvider = new Web3Provider(provider);
-    this._currentNetwork = networkController.store.getState().network;
```

Comment:
I removed these lines because they seemed unnecessary. The provider object should never change, so we should not need to recreate the wrapper around the provider.

Reply:
These lines are necessary because Ethers doesn't recognize when the network changes. The provider stops working. That's what we found when this was implemented at least.

Reply:
Not sure how I didn't notice this comment before. I'll make sure to revert this.

Reply:
I've brought back the previous version of this controller, making tweaks to ensure that we get network ID and status from the controller properly.